lib/surface/api.ex

```diff
@@ -192,9 +201,7 @@ defmodule Surface.API do
   end
 
   defp quoted_prop_funcs(env) do
-    props =
-      (Module.get_attribute(env.module, :prop) || [])
-      |> Enum.sort_by(&{&1.name != :id, !&1.opts[:required], &1.line})
```

Comment:
I haven't found a reason why the props are sorted like this. Tests are still going to pass even if we remove the sorting, except for the `Hint` inside a warning, so I moved the logic of sorting the props to the function that generate the warning message.

If there are any reasons why we have to sort props, I'll be glad to revert this change.

Reply:
Hmm. We should have a simple test for it.

This is the order props are listed in the docs. Example:

![image](https://user-images.githubusercontent.com/457237/111207213-151aa600-85a8-11eb-8040-42dbfba3aee7.png)

So first, we have all the required props and then the rest, all keeping the order they were defined. This way components authors have a simple way to organize the props, grouping them semantically. Not sure if it's the best way but it's what we have for now. I think we should keep it unless anyone has a better idea.

Reply:
Ok, it makes sense to order the props. From what I have seen from other tools (Vuetify, Sotrybook, Material-Ui) the API is always sorted by name and by category (props, slots, events, ...). If the prop/slot is required, the API shows a `*` or __required__. 

I suggest keeping the sorting as is. If we have feedback later, it will be easy to define a new sorting method. I'll revert these changes.

Reply:
I reverted the changes and I extracted the sorting into its function.